Userland/Applications/Calculator/Calculator.h

```diff
@@ -36,20 +37,30 @@ class Calculator final {
         MemClear,
         MemRecall,
         MemSave,
-        MemAdd
+        MemAdd,
+
+        Equals
     };
 
-    Crypto::BigFraction begin_operation(Operation, Crypto::BigFraction);
-    Crypto::BigFraction finish_operation(Crypto::BigFraction);
+    Optional<Crypto::BigFraction> operation_with_literal_argument(Operation, Crypto::BigFraction);
+    Optional<Crypto::BigFraction> operation_without_argument(Operation);
 
     bool has_error() const { return m_has_error; }
 
     void clear_operation();
     void clear_error() { m_has_error = false; }
 
 private:
-    Operation m_operation_in_progress { Operation::None };
-    Crypto::BigFraction m_saved_argument {};
     Crypto::BigFraction m_mem {};
+
+    Crypto::BigFraction m_current_value {};
+
+    Operation m_binary_operation_in_progress { Operation::None };
+    Crypto::BigFraction m_binary_operation_saved_left_side {};
+
+    Operation m_previous_operation { Operation::None };
+    Crypto::BigFraction m_previous_binary_operation_right_side {};
     bool m_has_error { false };
+
+    Crypto::BigFraction finish_binary_operation(Crypto::BigFraction left_side, Operation operation, Crypto::BigFraction right_side);
```

Comment:
`Crypto::BigFraction` can be huge, let's avoid a copy and take a `const&` here

Reply:
Done